### ISSUE TITLE
AeroSpace placing non-active workspace windows in the wrong bottom corner

### ISSUE THREAD

**Issue by mcjo163** [nikitabobko/AeroSpace#413]
First off, thanks so much for this great tool. It's my first time using a TWM and I can already feel the productivity boost!

I have the following monitor arrangement which I believe meets the guidelines [here](https://nikitabobko.github.io/AeroSpace/guide#proper-monitor-arrangement):
![image](https://github.com/user-attachments/assets/dda7c1e3-cbd4-4070-ad7f-49cbaf2d9588)

The issue is that non-active workspace windows on my center monitor are being placed in the bottom right corner and are showing up along the bottom of the right monitor. If I'm not mistaken, they should be placed in the bottom left corner of the center monitor, since it hangs below the left monitor there.

Screenshot from right monitor (the WezTerm window is in a workspace on my middle monitor, but I switched off that workspace):
![image](https://github.com/user-attachments/assets/2abc2221-f702-4779-ad05-c9cc8c0a38a2)

My config is basically the default with a couple of tweaked keybinds and `on-window-detected` callbacks, so I don't believe it's a configuration issue.

So my main question is "why is AeroSpace choosing the bottom right corner when the bottom left is the better choice?", and/or is there a way to force it to use a particular corner?

Please let me know if any other information would be helpful to provide!

## Checklist

- [x] I've tried the following queries to search for duplicates (please don't forget to search in closed issues too):
  - https://github.com/nikitabobko/AeroSpace/issues?q=is:issue+is:open+corner
  - https://github.com/nikitabobko/AeroSpace/issues?q=is:issue+is:open+monitor
- [x] I've read the documentation https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands & https://nikitabobko.github.io/AeroSpace/config-examples
- [x] I've tried searching for the following queries in the documentation:
  - "monitor" (I believe my arrangement meets the guidelines)

```
aerospace CLI client version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
AeroSpace.app server version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
```

**Comment by nikitabobko** [comment]
Right now, AeroSpace checks that no other monitor is presented in the bottom 10% of the monitor height. 10% is a very rough number and the algorithm can definetely be improved

The point being, is that AeroSpace thinks that both of your corners are bad corners, your temporary workaround is to lift the left monitor up

**Comment by mcjo163** [comment]
Aha, very good to know. Thanks!

**Comment by mcjo163** [comment]
So, further observation: I have rearranged like so...
<img width="253" alt="image" src="https://github.com/user-attachments/assets/654a6e3a-3c3a-447f-b72e-2b975063884d">

And my window appears across the bottom of my left monitor.
<img width="1099" alt="image" src="https://github.com/user-attachments/assets/54847d55-fe65-4e4a-aed5-4310a6825bb7">

I have tried with "Displays have separate spaces" on and off, both have the same result.

Any suggestions? Thanks.

**Comment by nikitabobko** [comment]
Can you please try the following custom build? [AeroSpace-v0.0.0-SNAPSHOT.zip](https://github.com/user-attachments/files/16619348/AeroSpace-v0.0.0-SNAPSHOT.zip)

It's built from 8cd4ddc07211b4aa6306a6e207798d10810e17f8 so alternatively you can also build from sources

**Comment by mcjo163** [comment]
That fixes the followup issue!

**Comment by nikitabobko** [comment]
0.14.2 is released to address the followup issue

The original issue is yet to be fixed

**Comment by mcjo163** [comment]
Wonderful, very much appreciated!

**Comment by jacobrreed** [comment]
@nikitabobko  
![image](https://github.com/user-attachments/assets/7055c47d-4c08-4679-8b68-0ef16cfe389e)
I seem to maybe have the same issue, I have my monitors arranged like so, monitor 1 has free space on left and monitor 2 has freespace on right. But I am seeing window flashing in bottom left corner of monitor 1
![image](https://github.com/user-attachments/assets/f020deab-a84c-440e-97ad-e47652117dc9)

**Comment by jacobrreed** [comment]
I was able to fix by moving right monitor up a little, although not ideal 
![image](https://github.com/user-attachments/assets/fff8e966-b2cc-47ce-98c5-2dc6048e39c7)

**Comment by sadid** [comment]
I'm currently trying to understand how aerospace handles screen boundaries. Similarly, I encountered the same issue with a different resolution. As a workaround, rearranging the monitors as suggested here works.
 
1920x1800
1800x1169 (Bult-in Display)
1920s1080

<img width="590" alt="Screenshot 2024-12-26 at 9 13 09 AM" src="https://github.com/user-attachments/assets/6980ee67-6c90-4823-baf5-88f96ee97357" />

**Comment by Motaphe** [comment]
It looks like the windows are showing up on the monitor where the dock is located. If you adjust that display to be slightly higher than your main display, it seems to _fix_ the issue.

![Image](https://github.com/user-attachments/assets/b84eba97-d46d-47a1-8640-6e86b9db7104)

**Comment by juanibiapina-ctfl** [comment]
This is my setup that has the same issue. The monitor on the right displays windows on the bottom part. It seems to use exactly the space below the main monitor. It's also interesting that this doesn't happen with a similar setup I have at home with the same mac but different monitors, cables and resolutions.

Is it possible that aerospace is also checking 10% of window width? That could explain the bug.

<img width="209" height="172" alt="Image" src="https://github.com/user-attachments/assets/978f55c8-8302-4f8e-8dd9-1ee02f6f6f5d" />

**Comment by juanibiapina-ctfl** [comment]
Indeed, if I reduce the size of the bottom display, the bug is gone. This matches my home setup.

Would it be possible to change the 10% threshold to something smaller, or make it configurable?

<img width="209" height="172" alt="Image" src="https://github.com/user-attachments/assets/eee66050-d889-4021-bef7-676a40f153f1" />

**Comment by nikitabobko** [comment]
The _optimal hide corner_ was improved in 0.20.0-Beta and should recognize all listed monitor configurations in this issue
